app/controllers/idv/personal_key_controller.rb

```diff
@@ -38,6 +45,25 @@ def update
       redirect_to next_step
     end
 
+    def self.step_info
+      Idv::StepInfo.new(
+        key: :personal_key,
+        controller: self,
+        action: :show,
+        next_steps: [FlowPolicy::FINAL],
+        preconditions: ->(idv_session:, user:) do
+          return false unless idv_session.address_confirmed? || idv_session.phone_confirmed?
+
+          return false unless user.active_or_pending_profile
```

Comment:
idv_session has a method for the first check, and I think you can combine it with the second one. (Remove trailing && if you're not keeping the third clause)
```suggestion
           idv_session.phone_or_address_step_complete? && user.active_or_pending_profile &&
```